pkg/cmd/cli/cmd/logs.go

```diff
@@ -155,7 +157,18 @@ func (o OpenShiftLogsOptions) Validate() error {
 	if o.Options == nil {
 		return nil
 	}
-	// TODO: Validate our own options.
+	switch t := o.Options.(type) {
+	case *buildapi.BuildLogOptions:
+		if t.Previous && t.Version != nil {
+			return errors.New("cannot use both --previous and --version")
+		}
+	case *deployapi.DeploymentLogOptions:
+		if t.Previous && t.Version != nil {
+			return errors.New("cannot use both --previous and --version")
+		}
```

Comment:
Yes, we can, although I prefer to keep the validation of each object separate as they may diverge in the future.